src/dict/aard.cc

```diff
@@ -295,11 +295,7 @@ AardDictionary::AardDictionary( string const & id, string const & indexFile, vec
   // Read dictionary name
 
   idx.seek( sizeof( idxHeader ) );
-  vector< char > dName( idx.read< quint32 >() );
-  if ( dName.size() ) {
-    idx.read( &dName.front(), dName.size() );
-    dictionaryName = string( &dName.front(), dName.size() );
-  }
+  idx.readU32SizeAndData<>( dictionaryName );
```

Comment:
IMO, readU32SizeAndData deduced type from the parameter.
what is the deduced type from variable `dictionaryName`?

Reply:
The deduced type is `decltype(dictionaryName) -> std::string`.

This line can also be written as, but I find it unnecessary.

```
idx.readU32SizeAndData<std::string>( dictionaryName );
```

Reply:
is operate on string itself is safe? ,as I can remember ,the string is immutable.

Reply:
```
std::string readDataWithLength() {
    int length;
    read(reinterpret_cast<char*>(&length), sizeof(length)); // 读取整数长度

    std::vector<char> buffer(length);
    read(buffer.data(), length); // 读取指定长度的数据


    return std::string(buffer.begin(), buffer.end());
}
```

Reply:
std::string is mutable like std::vector.

Reply:
```cpp
template< typename T >
std::string readDataWithLength() {
```
I tried that a few months ago, https://github.com/xiaoyifang/goldendict-ng/pull/1435/files, and it doesn't work.

Type name cannot be automatically deduced from return type.

I cannot really explain why now, but put the to-be-muted vector/string in parameter looks to be simple, because we can just do `do<>(some_obect);`.